sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala

```diff
@@ -438,23 +438,25 @@ class RelationalGroupedDataset protected[sql](
 
   private[sql] def flatMapGroupsInPandas(expr: PythonUDF): DataFrame = {
     require(expr.vectorized, "Must pass a vectorized python udf")
+    require(expr.dataType.isInstanceOf[StructType],
+      "The returnType of the vectorized python udf must be a StructType")
 
-    val output = expr.dataType match {
-      case s: StructType => s.map {
-        case StructField(name, dataType, nullable, metadata) =>
-          AttributeReference(name, dataType, nullable, metadata)()
-      }
+    val groupingNamedExpressions = groupingExprs.map {
```

Comment:
Just wondering if this can be `groupingExprs.map(alias)` (looks it is in `flatMapGroupsInR`).

Reply:
I might miss something, but it seems that the `alias` sometimes contains `UnresolvedAlias`, so the following `toAttribute` fails.
It works if calling `toAttribute` after resolved, but doesn't work in this case.

Reply:
Do you mind explaining a bit more? `Alias(NamedExpression).toAttribute` doesn't work here but `Alias(NonNamedExpression).toAttribute` works? Why?

Reply:
The `alias` we say here is `RelationalGroupedDataset.alias(expr: Expression): NamedExpression` method. It can be used to make `NamedExpression` from `Expression` but it sometimes returns `UnresolvedAlias`.